src/test/java/ProjectOcean/WorkspaceTest.java

```diff
@@ -1,106 +1,168 @@
 package ProjectOcean;
 
 import ProjectOcean.Model.Course;
+import ProjectOcean.Model.CourseFactory;
+import ProjectOcean.Model.ICourse;
 import ProjectOcean.Model.Workspace;
 import org.junit.Assert;
+import org.junit.Before;
 import org.junit.Test;
 
 import java.util.LinkedList;
 import java.util.List;
 
 public class WorkspaceTest {
 
-    @Test
-    public void getAllCoursesTest(){
-        Workspace ws = new Workspace();
-        Course course = new Course("BAT123", "Beroendespecifika paradigmer", 7.5F,3,"Anders Bölinge", "Tenta","Turkish", null, "LINK_Zelda", "Abu Dhabi. YOLO!");
-        Course course2 = new Course("CAT123", "Complex system", 7.5F, 2, "Anders Fölinge", "Tenta","Turkish", null, "LINK_Zelda", "Abu Dhabi. YOLO!");
-        Course course3 = new Course("DAT321", "Datavetenskap", 7.5F, 4, "Anders Bölinge", "Tenta","Turkish", null, "LINK_Zelda", "Abu Dhabi. YOLO!");
-        Course course4 = new Course("FAT321", "Fysik för ingenjörer", 7.5F, 1, "Anders Brölinge", "Tenta","Turkish", null, "LINK_Zelda", "Abu Dhabi. YOLO!");
+    private Workspace ws = new Workspace();
+    private List courses = new LinkedList<Course>();
+
+    @Before
+    public void createCourses() {
+        CourseFactory.SetStudyPeriod(3);
+        CourseFactory.SetCourseInfo("BAT123", "Beroendespecifika paradigmer", 7.5F);
+        CourseFactory.SetCourseDetails(null, "LINK_Zelda", "Abu Dhabi. YOLO!");
+        CourseFactory.SetCourseAccessibility("Anders Bölinge", "Tenta","Turkish");
+        ICourse course = CourseFactory.CreateCourse();
+
+        CourseFactory.SetStudyPeriod(2);
+        CourseFactory.SetCourseInfo("CAT123", "Complex system", 7.5F);
+        CourseFactory.SetCourseDetails(null, "LINK_Zelda", "Abu Dhabi. YOLO!");
+        CourseFactory.SetCourseAccessibility("Anders Fölinge", "Tenta","Turkish");
+        ICourse course2 = CourseFactory.CreateCourse();
+
+        CourseFactory.SetStudyPeriod(4);
+        CourseFactory.SetCourseInfo("DAT321", "Datavetenskap", 7.5F);
+        CourseFactory.SetCourseDetails(null, "LINK_Zelda", "Abu Dhabi. YOLO!");
+        CourseFactory.SetCourseAccessibility("Anders Bölinge", "Tenta","Turkish");
+        ICourse course3 = CourseFactory.CreateCourse();
+
+        CourseFactory.SetStudyPeriod(1);
+        CourseFactory.SetCourseInfo("FAT321", "Fysik för ingenjörer", 7.5F);
+        CourseFactory.SetCourseDetails(null, "LINK_Zelda", "Abu Dhabi. YOLO!");
+        CourseFactory.SetCourseAccessibility("Anders Brölinge", "Tenta","Turkish");
+        ICourse course4 = CourseFactory.CreateCourse();
```

Comment:
Dont know if this will be reviewed by Alex, but if it will, this is not very proffessional?